src/simulation/dynamics/spinningBodies/spinningBodiesTwoDOF/spinningBodyTwoDOFStateEffector.rst

```diff
@@ -0,0 +1,124 @@
+
+Executive Summary
+-----------------
+
+The two-degree-of-freedom spinning body class is an instantiation of the state effector abstract class. The integrated test is validating the interaction between the 2-DoF spinning body module and the rigid body hub that it is attached to. In this case, a 2-DoF spinning body system has two masses and two inertia tensors. The lower axis is attached to the hub and the upper axis is attached to the lower body. This module can represent multiple different effectors, such as a dual-hinged solar array, a control moment gyroscope or a dual-gimbal. A spring and damper can be included in each axis, and an optional motor torque can be applied on each spinning axis.
+
+Nominally, two rigid bodies can be defined, which would represent a chain of 1D rotary joints. However, by setting the mass and the inertia of the lower spinning body to 0, the module can simulate a single rigid component rotating about two axis instead.
+
+
+Message Connection Descriptions
+-------------------------------
+The following table lists all the module input and output messages.  The module msg variable name is set by the user from python.  The msg type contains a link to the message structure definition, while the description provides information on what this message is used for.
+
+.. list-table:: Module I/O Messages
+    :widths: 25 25 50
+    :header-rows: 1
+
+    * - Msg Variable Name
+      - Msg Type
+      - Description
+    * - spinningBodyOutMsgs
+      - :ref:`HingedRigidBodyMsgPayload`
+      - Output vector of messages containing the spinning body state angle and angle rate
+    * - motorTorqueInMsg
+      - :ref:`ArrayMotorTorqueMsgPayload`
+      - (Optional) Input message of the motor torque value
+    * - motorLockInMsg
+      - :ref:`ArrayEffectorLockMsgPayload`
+      - (Optional) Input message for locking the axis
+    * - spinningBodyConfigLogOutMsgs
+      - :ref:`SCStatesMsgPayload`
+      - Output vector of messages containing the spinning body inertial position and attitude states
+
+
+Detailed Module Description
+---------------------------
+
+A 2 DoF spinning body has 4 states: ``theta1``, ``theta2``, ``theta1Dot`` and ``theta2Dot``.
+
+Mathematical Modeling
+^^^^^^^^^^^^^^^^^^^^^
+See the following conference paper for a detailed description of this model.
+
+.. note::
+
+    J. Vaz Carneiro, C. Allard and H. Schaub, “Rotating Rigid Body Dynamics
+    Architecture for Spacecraft Simulation Software Implementation,” AAS Rocky
+    Mountain GN&C Conference, Breckenridge, CO, Feb. 2–8, 2023 <https://hanspeterschaub.info/Papers/VazCarneiro2023.pdf>
```

Comment:
make the title the link text instead of showing the full url path?  Could use

“`Rotating Rigid Body Dynamics
    Architecture for Spacecraft Simulation Software Implementation<https://hanspeterschaub.info/Papers/VazCarneiro2023.pdf>`__,”

Reply:
That's a great idea!